From 2ee08ffd4d469781dc627fa50b4a015d9ad68007 Mon Sep 17 00:00:00 2001
From: Ludovico de Nittis <ludovico.denittis@collabora.com>
Date: Fri, 10 Jan 2025 18:31:46 +0100
Subject: [PATCH] device: Clear pending_wake_allowed on error

If setting WakeAllowed fails, we should unset the `pending_wake_allowed`
property, or subsequent requests will be rejected due to another
"Property change in progress".

Fixes: https://github.com/bluez/bluez/issues/1047
---
 src/device.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/device.c b/src/device.c
index 24ef3d779..e8bff718c 100644
--- a/src/device.c
+++ b/src/device.c
@@ -1574,6 +1574,7 @@ static void set_wake_allowed_complete(uint8_t status, uint16_t length,
 						      mgmt_errstr(status));
 			dev->wake_id = -1U;
 		}
+		dev->pending_wake_allowed = FALSE;
 		return;
 	}
 
